node/follower_node.go

```diff
@@ -135,10 +135,10 @@ func MakeFollower(log logging.Logger, rootDir string, cfg config.Local, phoneboo
 	node.catchupBlockAuth = blockAuthenticatorImpl{Ledger: node.ledger, AsyncVoteVerifier: agreement.MakeAsyncVoteVerifier(node.lowPriorityCryptoVerificationPool)}
 	node.catchupService = catchup.MakeService(node.log, node.config, p2pNode, node.ledger, node.catchupBlockAuth, make(chan catchup.PendingUnmatchedCertificate), node.lowPriorityCryptoVerificationPool)
 
-	// Initialize sync round to the next round so that nothing falls out of the cache on Start
-	err = node.SetSyncRound(uint64(node.Ledger().NextRound()))
+	// Initialize sync round to the latest db round + 1 so that nothing falls out of the cache on Start
+	err = node.SetSyncRound(uint64(node.Ledger().DBRound() + 1))
```

Comment:
I'm not sure we want the `+ 1`. Hopefully the test can help determine the answer.

Reply:
The tests depend on this functionality. Note that getting a sync round from algod with value 0 [will error](https://github.com/algorand/go-algorand/blob/b7234a38eeba64a5114261a44caffefaac6e82d1/daemon/algod/api/server/v2/handlers.go#L1087-L1088) as it's considered to be "unset", so that calling the endpoint immediately after init would error currently, without the `+1`.

I'm viewing the sync round as a "strict upper bound" rather than being "inclusive".

However, if this is fundamentally changing the definition of sync round along with expected behavior, then we should remove the `+1`. In that case, I would argue that we also need to change the `/v2/ledger/sync` endpoint to distinguish between the sync round being unset and sync round being initialized to 0.